### ISSUE TITLE
 formControl value changes or validators stops working after navigating to second the page .

### ISSUE THREAD

**Issue by chaithrab12** [NativeScript/NativeScript#8748]
**Environment**

- Cross-platform modules: "@nativescript/core": "rc", "@nativescript/angular": "~10.0.0", "@nativescript/webpack": "^2.0.0"

*Describe the bug**

I have angular 10 project migrated to nativescript latest version. I have a bug with formControl .When I load the application ,In the first page formControl works .The values changes are detected ,When I navigate to next page formControl doesn't detect changes on TextField or on TextView. I tried using _textChange_  event even thats not getting triggered  on the second page . 
I also used [(ngModel)] ,this too doesn't work.
It doesn't even throw any error .
tsconfig.json
```
{
  "compileOnSave": false,
  "compilerOptions": {
    "baseUrl": "./",
    "outDir": "./dist/out-tsc",
    "sourceMap": true,
    "declaration": false,
    "downlevelIteration": true,
    "experimentalDecorators": true,
    "moduleResolution": "node",
    "importHelpers": true,
    "target": "es2015",
    "module": "esnext",
    "lib": [
      "es2018",
      "dom"
    ],
    "paths": {
      "@src/*": [
        "src/*.android.ts",
        "src/*.ios.ts",
        "src/*.tns.ts",
        "src/*.web.ts",
        "src/*"
      ]
    }
  }
}
```

tsconfig.tns.json

```
{
  "extends": "./tsconfig.json",
  "compilerOptions": {
    "module": "ESNext",
    "moduleResolution": "node",
    "experimentalDecorators": true,
    "skipLibCheck": true,
    "baseUrl": ".",
    "paths": {
      "@src/*": [
        "src/*.tns.ts",
        "src/*.ts"
      ]
    }
  },
  "files": [
    "src/main.tns.ts"
  ]
}
```

Package.json
```

{
  "name": "angularapp-v10",
  "version": "0.0.0",
  "scripts": {
    "android": "tns run android --no-hmr",
    "ios": "tns run ios --no-hmr",
    "mobile": "tns run",
    "preview": "tns preview",
    "ngcc": "ngcc --properties es2015 module main --first-only",
    "postinstall": "npm run ngcc",
    "ng": "ng",
    "start": "ng serve",
    "build": "ng build",
    "test": "ng test",
    "lint": "ng lint",
    "e2e": "ng e2e"
  },
  "private": true,
  "dependencies": {
    "@nativescript/angular": "~10.0.0",
    "@nativescript/core": "rc",
    "@nativescript/theme": "~2.2.1",
    "reflect-metadata": "~0.1.12",
    "tslib": "^2.0.0",
    "@angular/animations": "~10.0.6",
    "@angular/common": "~10.0.6",
    "@angular/compiler": "~10.0.6",
    "@angular/core": "~10.0.6",
    "@angular/forms": "~10.0.6",
    "@angular/platform-browser": "~10.0.6",
    "@angular/platform-browser-dynamic": "~10.0.6",
    "@angular/router": "~10.0.6",
    "rxjs": "~6.5.5",
    "zone.js": "~0.10.3"
  },
  "devDependencies": {
    "@angular-devkit/build-angular": "~0.1000.5",
    "@angular/cli": "~10.0.5",
    "@angular/compiler-cli": "~10.0.6",
    "@nativescript/schematics": "^10.0.2",
    "@nativescript/tslint-rules": "~0.0.5",
    "@nativescript/webpack": "~2.0.0",
    "@types/jasmine": "~3.5.0",
    "@types/jasminewd2": "~2.0.3",
    "@types/node": "^12.11.1",
    "codelyzer": "^6.0.0",
    "jasmine-core": "~3.5.0",
    "jasmine-spec-reporter": "~5.0.0",
    "karma": "~5.0.0",
    "karma-chrome-launcher": "~3.1.0",
    "karma-coverage-istanbul-reporter": "~3.0.2",
    "karma-jasmine": "~3.3.0",
    "karma-jasmine-html-reporter": "^1.5.0",
    "nativescript-dev-webpack": "^1.5.1",
    "protractor": "~7.0.0",
    "ts-node": "~8.3.0",
    "tslint": "~6.1.0",
    "typescript": "~3.9.5"
  },
  "nativescript": {
    "id": "org.nativescript.ngsample",
    "tns-ios": {
      "version": "6.5.2"
    }
  }
}
```
app.module.tns.ts

```
import { NgModule, NO_ERRORS_SCHEMA } from '@angular/core';
import { NativeScriptModule, NativeScriptFormsModule, NativeScriptCommonModule } from '@nativescript/angular';
import { ReactiveFormsModule } from '@angular/forms';
import { AppRoutingModule } from '@src/app/app-routing.module.tns';
import { AppComponent } from '@src/app/app.component';
import { AutoGeneratedComponent } from '@src/app/auto-generated/auto-generated.component';
import { ItemComponent } from '@src/app/item/item.component';

@NgModule({
  declarations: [
    AppComponent,
    AutoGeneratedComponent,
    ItemComponent,
  ],
  imports: [
    NativeScriptModule,
    NativeScriptFormsModule,
    ReactiveFormsModule,
    AppRoutingModule,
  
  ],
  providers: [],
  bootstrap: [AppComponent],
  schemas: [NO_ERRORS_SCHEMA]
})
/*
Pass your application module to the bootstrapModule function located in main.ts to start your app
*/
export class AppModule { }
```

Using this template

```
    <StackLayout [formGroup]="formLogin">
        <TextField formControlName="emailAddress" hint="add text"></TextField>
        <Button text="Click" (tap)="click()"></Button>
    </StackLayout>
```
ts file 

  ```
 ngOnInit() {
  this.formLogin = this.formBuilder.group({
      emailAddress:new FormControl('', [Validators.required]),
    });
  }

  public click(){
  console.log(this.formLogin.value.emailAddress);
    this.router.navigate(['/item']);
  }

```

**Comment by spencerfontein** [comment]
@NathanWalker I know you are busy with lots of Angular 10 things, but I would really appreciate an update on this.

**Comment by NathanWalker** [comment]
@spencerfontein is that a sample you can drop on github just so I can run the exact setup? That would help me dig into the exact case as I tried setting up something similar but was unable to repro an issue as described - I could be missing something with the setup that leads to it?

**Comment by chaithrab12** [comment]
@NathanWalker I have dropped in the sample code on my github  where you can repro the from control issue .[https://github.com/chaithrab12/Nativescript-angular-app]. 
Thank you

**Comment by spencerfontein** [comment]
@NathanWalker have you had a chance to to review the sample, we are still having issues?

**Comment by NathanWalker** [comment]
@spencerfontein @chaithrab12 I have been looking at this - this has something to do with the zone overrides that the {N} team had put in the angular integration back at beginning on Angular 2 (2017ish) -- Seems zone change detection is affected on that navigation - I'll let you know if find a fix/solution or tmp workaround - def curious.
This is one of the reasons we plan to rewrite the angular integration from scratch for angular 11 but I'm sure we can find a root cause here.

**Comment by spencerfontein** [comment]
@NathanWalker thank you for the update. Have you been able to find a fix or workaround, really appreciate your help with this.

**Comment by NathanWalker** [comment]
@spencerfontein We did start the rewrite however started digging back into this case today and another contributor is helping to look into this case over weekend. We definitely want to provide some answers here by Monday.

**Comment by briandilley** [comment]
We're having the same issue with angular 7.  It doesn't happen after every navigation, but navigating between two screens (register/login) repeatedly eventually breaks it and navigation does nothing (it doesn't seem to be calling the navigation function anymore).  Also - two-way binding works "sometimes" (via `[(ngModel]]`).  By "somtimes" i mean it works 75% of the time on the registration screen, and basically never on the login screen... and the code is basically identical.  I feel like this all might be related (perhaps events aren't being fired anymore and that's why it's not updating the model or navigating?).  Here's our shinizle:

tsconfig.json
```
{
  "compilerOptions": {
    "module": "esnext",
    "target": "es2017",
    "moduleResolution": "node",
    "experimentalDecorators": true,
    "emitDecoratorMetadata": true,
    "noEmitHelpers": true,
    "noEmitOnError": true,
    "skipLibCheck": true,
    "lib": [
      "es2018",
      "es2017",
      "dom",
      "es6"
    ],
    "baseUrl": ".",
    "paths": {
      "~/*": [
        "app/*"
      ]
    },
    "removeComments": false
  },
  "include": [
    "src/**/*.ts"
  ],
  "files": [
    "./references.d.ts",
    "./nativescript.config.ts"
  ],
  "exclude": [
    "node_modules",
    "platforms",
    "e2e"
  ]
}
```

nativescript.config.ts
```
import { NativeScriptConfig } from '@nativescript/core'

export default {
  id: 'com.getxxx.xxx',
  appResourcesPath: 'App_Resources',
  android: {
    v8Flags: '--expose_gc',
    markingMode: 'none',
  },
  appPath: 'src',
} as NativeScriptConfig
```

app.module.ts
```
import {NgModule, NO_ERRORS_SCHEMA} from "@angular/core";
import {
    NativeScriptCommonModule,
    NativeScriptFormsModule,
    NativeScriptModule
} from "@nativescript/angular";

import {AppRoutingModule} from "./app-routing.module";
import {AppComponent} from "./app.component";
import {RegisterComponent} from "./register/register.component";
import {LoginComponent} from "./login/login.component";
import {StartupComponent} from "./startup/start.component";
import {LibModule} from "../lib/lib.module";
import {MainComponent} from "./main/main.component";
import {IntroComponent} from "./intro/intro.component";

@NgModule({
    bootstrap: [
        AppComponent
    ],
    imports: [
        NativeScriptModule,
        NativeScriptCommonModule,
        NativeScriptFormsModule,
        AppRoutingModule,
        LibModule
    ],
    declarations: [
        AppComponent,
        IntroComponent,
        StartupComponent,
        RegisterComponent,
        LoginComponent,
        MainComponent
    ],
    providers: [],
    schemas: [
        NO_ERRORS_SCHEMA
    ]
})
export class AppModule { }
```

package.json
```
{
  "name": "@nativescript/xxx",
  "main": "main.js",
  "version": "7.0.2",
  "author": "XXX Team <engineering@getxxx.com>",
  "description": "XXX NativeScript App",
  "publishConfig": {
    "access": "private"
  },
  "keywords": [
    "xxx"
  ],
  "repository": "git@github.com:XXXEngineering/xxx-mobile-nativescript.git",
  "bugs": {
    "url": "https://github.com/XXXEngineering/xxx-mobile-nativescript/issues"
  },
  "scripts": {
    "ngcc": "ngcc --properties es2015 module main --first-only",
    "postinstall": "npm run ngcc",
    "lint": "eslint 'src/**/*.{js,ts,tsx}'",
    "lint-watch": "esw 'src/**/*.{js,ts,tsx}' --watch-delay 1000 --watch --clear --color"
  },
  "dependencies": {
    "@angular/animations": "~10.1.0",
    "@angular/common": "~10.1.0",
    "@angular/compiler": "~10.1.0",
    "@angular/core": "~10.1.0",
    "@angular/forms": "~10.1.0",
    "@angular/platform-browser": "~10.1.0",
    "@angular/platform-browser-dynamic": "~10.1.0",
    "@angular/router": "~10.1.0",
    "@nativescript/angular": "~10.1.0",
    "@nativescript/core": "~7.0.0",
    "@nativescript/theme": "~2.3.0",
    "reflect-metadata": "~0.1.12",
    "rxjs": "^6.6.0",
    "zone.js": "~0.11.1"
  },
  "devDependencies": {
    "@angular/compiler-cli": "~10.1.0",
    "@nativescript/android": "7.0.0",
    "@nativescript/ios": "7.0.0",
    "@nativescript/types": "~7.0.0",
    "@nativescript/webpack": "~3.0.0",
    "@ngtools/webpack": "~10.1.0",
    "@typescript-eslint/eslint-plugin": "^4.0.1",
    "@typescript-eslint/parser": "^4.0.1",
    "eslint": "^7.8.1",
    "eslint-loader": "^4.0.2",
    "eslint-watch": "^7.0.0",
    "node-sass": "^4.14.1",
    "typescript": "~3.9.0"
  },
  "private": "true",
  "readme": "XXX NativeScript App"
}

```

this is how we're navigating:
```
        <Button text="I have an account, Sign In" (tap)="gotoLogin()" class="link"></Button>

        public gotoLogin(): void {
        this.router.navigateByUrl("/login", {
            animated: true,
            clearHistory: true,
            transition: {
                name: "slideright",
                duration: 300
            }
        });
    }
```

and this is how we're binding fields:
```
        <TextField hint="Name" [(ngModel)]="name" autocorrect="false" autocapitalizationType="words" maxLength="32">
        </TextField>

        <TextField hint="Email address" [(ngModel)]="emailAddress" keyboardType="email" autocorrect="false"
            autocapitalizationType="none" maxLength="128">
        </TextField>

        <TextField hint="Password" [(ngModel)]="password" secure="true" minlength="5" maxLength="64">
        </TextField>

export class RegisterComponent {

    public name: string = "";
    public emailAddress: string = "";
    public password: string = "";
```

**Comment by spencerfontein** [comment]
Hello @NathanWalker,

Thank you for you and the other contributors looking into this, we tried disabling Ivy and also upgraded to latest Nativescript 7 and the problem still persists. Please let me know we can provide any more details and let us know if you have any updates.

Thank you
Spencer

**Comment by NathanWalker** [comment]
This issue is now fixed with @nativescript/angular 10.1.4 published now 👍

**Comment by briandilley** [comment]
> This issue is now fixed with @NativeScript/angular 10.1.4 published now 👍

what was the issue?

**Comment by NathanWalker** [comment]
@briandilley Ivy exposed some underlying issues with the DetachedLoaded reference handling when moving visual nodes around the component tree which ultimately was leading to losing delegates on iOS. That is now using best practices as used in angular/cdk itself and fixed by @edusperoni here in the integration 👍
